dbms/src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -523,17 +525,51 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
               * Otherwise, concurrent queries for creating a table, if the table does not exist,
               *  can throw an exception, even if IF NOT EXISTS is specified.
               */
-            guard = context.getDDLGuardIfTableDoesntExist(
-                database_name,
-                table_name,
-                "Table " + database_name + "." + table_name + " is creating or attaching right now");
+            try
+            {
+                guard = context.getDDLGuardIfTableDoesntExist(
```

Comment:
I think the DDLGuard::map is clean up under the destructor of DDLGuard. So when a ddl_gurad is at the end of its life cycle, it will erase itself from DDLGuard::map.

for the database -> table -> exception_message, we can simplify it later.

Reply:
I think it can only remove items from `table -> exception_message`. Anyway, we don't need to optimize it in this PR. Can you file an issue and address it later?

Reply:
Of course, I create a new issue(https://github.com/pingcap/tiflash/issues/7636) and list it in https://github.com/pingcap/tiflash/issues/7630